### PR TITLE
feat: Make  year optional query parameter

### PR DESCRIPTION
**Pull request by LarsV123**
Makes the  query parameter `year` optional for journal, series, and publication searches.

Refs: https://sikt.atlassian.net/browse/NP-46251

### CODE REVIEW

**Review by truhacevkir — APPROVED** [review]
Nice, it seems like no one of tests is testing if id of paginated search result is getting expected value.

If we run following assertion on a random test, it will fail:
`assertThat(pagesSearchResult.getId(),is(equalTo(expectedSearchResult.getId())));`

Do not know how important it is, but i think we should have at least one test which tests uri/id returned in response :)

File: src/main/java/no/sikt/nva/pubchannels/handler/search/SearchByQueryHandler.java [hunk 1]
```diff
@@ -79,13 +79,18 @@ protected PaginatedSearchResult<T> processInput(Void input, RequestInfo requestI
...
+        baseQueryParameters.remove(PAGECOUNT_QUERY_PARAM);
+
         return PaginatedSearchResult.create(
             constructBaseUri(),
             searchParameters.offset(),
             searchParameters.size(),
             searchResult.pageInformation().totalResults(),
             getHits(constructBaseUri(), searchResult, searchParameters.year()),
-            Map.of(QUERY_PARAM, searchParameters.query(), YEAR_QUERY_PARAM, searchParameters.year())
+            baseQueryParameters
```
Comment (monaullah) [inline src/main/java/no/sikt/nva/pubchannels/handler/search/SearchByQueryHandler.java:93]: As mentioned in call, these query parameters are not correct input query params for this endpoint. I see that this is not covered in tests (unfortunately). Suggest writing tests asserting expected `id` and `query parameters` in `PaginatedSearchResult`.

**Review by monaullah — CHANGES_REQUESTED** [review]
See comment about query params in `PaginatedSearchResult`

File: src/main/java/no/sikt/nva/pubchannels/handler/search/SearchByQueryHandler.java (hunk 1 above)
Comment (LarsV123) [reply src/main/java/no/sikt/nva/pubchannels/handler/search/SearchByQueryHandler.java:95]: This should be resolved now, can you take another look?

**Review by monaullah — APPROVED** [review]
Nice!

**Review by truhacevkir — APPROVED** [review]

**Pull request merged**